packages/eslint-plugin/src/rules/prefer-as-const.ts

```diff
@@ -0,0 +1,84 @@
+import {
+  AST_NODE_TYPES,
+  TSESLint,
+  TSESTree,
+} from '@typescript-eslint/experimental-utils';
+import * as util from '../util';
+
+export default util.createRule({
+  name: 'prefer-as-const',
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description: 'Prefer usage of `as const` over literal type',
+      category: 'Best Practices',
+      recommended: false,
+    },
+    fixable: 'code',
+    messages: {
+      preferConstAssertion:
+        'Expected a `const` instead of a literal type assertion',
+      variableConstAssertion:
+        'Expected a `const` assertion instead of a literal type annotation',
+      variableSuggest1: 'You should add `as const` assertion.',
+      variableSuggest2: 'You should use `as const` instead of type annotation.',
+    },
+    schema: [],
+  },
+  defaultOptions: [],
+  create(context) {
+    function compareTypes(
+      valueNode: TSESTree.Expression,
+      typeNode: TSESTree.TypeNode,
+      canFix: boolean,
+    ): void {
+      if (
+        valueNode.type === AST_NODE_TYPES.Literal &&
+        typeNode.type === AST_NODE_TYPES.TSLiteralType &&
+        'raw' in typeNode.literal &&
+        valueNode.raw === typeNode.literal.raw
+      ) {
+        if (canFix) {
+          context.report({
+            node: typeNode,
+            messageId: 'preferConstAssertion',
+            fix: fixer => fixer.replaceText(typeNode, 'const'),
+          });
+        } else {
+          context.report({
+            node: typeNode,
+            messageId: 'variableConstAssertion',
+            suggest: [
+              {
+                messageId: 'variableSuggest1',
+                fix: (fixer): TSESLint.RuleFix =>
+                  fixer.insertTextAfter(valueNode, ' as const'),
+              },
```

Comment:
how come you have this fixer?
It seems weird to suggest fixing `const x: 2 = 2;` to `const x: 2 = 2 as const;`

Reply:
ok i removed this suggestion